polyfill/lib/zoneddatetime.mjs

```diff
@@ -441,7 +442,54 @@ export class ZonedDateTime {
   }
   toLocaleString(locales = undefined, options = undefined) {
     if (!ES.IsTemporalZonedDateTime(this)) throw new TypeError('invalid receiver');
-    return new DateTimeFormat(locales, options).format(this);
+    options = ES.GetOptionsObject(options);
+
+    const optionsCopy = ObjectCreate(null);
+    // This is not quite per specification, but this polyfill's DateTimeFormat
+    // already doesn't match the InitializeDateTimeFormat operation, and the
+    // access order might change anyway;
+    // see https://github.com/tc39/ecma402/issues/747
+    ES.CopyDataProperties(optionsCopy, options, ['timeZone']);
+
+    if (options.timeZone !== undefined) {
+      throw new TypeError('ZonedDateTime toLocaleString does not accept a timeZone option');
+    }
+
+    if (
+      optionsCopy.year === undefined &&
+      optionsCopy.month === undefined &&
+      optionsCopy.day === undefined &&
+      optionsCopy.weekday === undefined &&
+      optionsCopy.dateStyle === undefined &&
+      optionsCopy.hour === undefined &&
+      optionsCopy.minute === undefined &&
+      optionsCopy.second === undefined &&
+      optionsCopy.timeStyle === undefined &&
+      optionsCopy.dayPeriod === undefined &&
+      optionsCopy.timeZoneName === undefined
+    ) {
+      optionsCopy.timeZoneName = 'short';
+      // The rest of the defaults will be filled in by formatting the Instant
+    }
+
+    let timeZone = ES.ToTemporalTimeZoneIdentifier(GetSlot(this, TIME_ZONE));
+    if (ES.IsTimeZoneOffsetString(timeZone)) {
+      throw new RangeError('toLocaleString does not support offset string time zones');
```

Comment:
Sure, I'll add the TODO. I wouldn't recommend adding a workaround for offset time zones though, since this code is supposed to be a reference implementation for the spec.